File: src/com/peqnp/FactoringInterop.java
package com.peqnp;

import org.python.util.PythonInterpreter;

public class FactoringInterop {

    public static void main(String[] args) {
        PythonInterpreter py = new PythonInterpreter();
        py.set("rsa", 3007);
        py.execfile("py/factoring_interop.py");
        System.out.println(py.get("p"));
        System.out.println(py.get("q"));
    }
}

File: py/factoring_interop.py
import peqnp.cnf as cnf

# rsa set from java

cnf.begin(bits=rsa.bit_length(), key='factoring_interop')
p = cnf.integer()
q = cnf.integer()
assert p * q == rsa
cnf.end({'p': p, 'q': q})

cnf.satisfy(solver='java -jar -Xmx4g blue.jar')

print(p, q)